### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by AlexandruGG**
This PR updates the API documentation and README with details similar to other examples.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -236,7 +236,6 @@ For more detailed changelogs, please check [the release changelogs][45].
 [33]: https://loophp-collection.rtfd.io
 [28]: https://loophp-collection.readthedocs.io/en/latest/pages/api.html
 [32]: https://loophp-collection.readthedocs.io/en/latest/pages/usage.html
-[46]: https://loophp-collection.readthedocs.io/en/latest/pages/examples.html
```
Comment (AlexandruGG) [inline README.md:239]: This is a dead URL, I'm not sure if it pointed to a page that no longer exists or the page exists but it was just renamed, please advise.

File: docs/conf.py [hunk 2]
```diff
@@ -158,8 +158,8 @@
 
 
 def setup(app):
-    app.add_stylesheet('custom.css')
-    app.add_stylesheet('highlight.css')
```
Comment (AlexandruGG) [inline docs/conf.py:162]: when trying to preview the `.rst` files in VSCode it was failing with this error: https://github.com/sphinx-doc/sphinx/issues/7747. The change I made fixes it for me, is it ok to make? I'm not too familiar with Sphinx usage

File: docs/pages/api.rst [hunk 3]
```diff
@@ -435,6 +447,11 @@ Interface: `Containsable`_
 
 Signature: ``Collection::contains(...$value);``
 
+.. code-block:: php
+
+    $collection = Collection::fromIterable(range('a', 'c'))
+        ->contains('d'); // [false]
```
Comment (AlexandruGG) [inline docs/pages/api.rst:453]: this behaviour is based on the spec but I have to say it surprised me. Shouldn't the return type be just `bool`? I would expect to use this method as such:

```PHP

$collection = Collection::fromIterable([1, 2, 3]);

if ($collection->contains(2)) {
    // do something
}
```

How is the current implementation meant to be used?

File: docs/pages/api.rst [hunk 4]
```diff
@@ -454,16 +471,16 @@ Signature: ``Collection::current(int $index = 0);``
 cycle
 ~~~~~
 
-Cycle around a collection of items.
+Cycle indefinitely around a collection of items.
 
 Interface: `Cycleable`_
 
-Signature: ``Collection::cycle(int $length = 0);``
```
Comment (AlexandruGG) [inline docs/pages/api.rst:461]: I'm not sure if this was for an older version, but the current `cycle` method doesn't take a parameter and cycles indefinitely from what I can see

File: README.md (hunk 1 above)
Comment (drupol) [reply README.md:239]: Thanks!

File: docs/pages/api.rst (hunk 3 above)
Comment (drupol) [reply docs/pages/api.rst:453]: Indeed, this was a choice that I made in order to make sure that the collection would always return a collection.
The 3 only methods that are not potentially returning a Collection are: `all`, `key`, `current`.
In this example, to get the boolean, you have to do:

```php
$collection = Collection::fromIterable([1, 2, 3]);

if ($collection->contains(2)->current()) {
    // do something
}
```

File: docs/pages/api.rst (hunk 4 above)
Comment (drupol) [reply docs/pages/api.rst:461]: Seems like this was from an old implementation. Maybe we should keep the signature, but just updating it:

` Signature: ``Collection::cycle();`` `

**Review by drupol — APPROVED** [review]
Ignore this approval comment, I hit the wrong button!

**Review by drupol — CHANGES_REQUESTED** [review]
Thanks!

I made a review. All those changes are great!

There's a small things to update, as soon as it's ok, I merge this.

File: docs/pages/api.rst [hunk 5]
```diff
@@ -158,6 +158,18 @@ Interface: `Allable`_
 
 Signature: ``Collection::all();``
 
+.. code-block:: php
+
+        $generator = static function (): Generator {
+            yield 0 => 'a';
+            yield 1 => 'b';
+            yield 0 => 'a';
```
Comment (drupol) [inline docs/pages/api.rst:166]:
```suggestion
            yield 0 => 'c';
```

File: docs/pages/api.rst [hunk 6]
```diff
@@ -158,6 +158,18 @@ Interface: `Allable`_
...
+            yield 2 => 'c';
```
Comment (drupol) [inline docs/pages/api.rst:167]:
```suggestion
            yield 2 => 'd';
```

File: docs/pages/api.rst [hunk 7]
```diff
@@ -158,6 +158,18 @@ Interface: `Allable`_
...
+        };
+
+        Collection::fromIterable($generator())
+            ->all(); // [0 => 'a', 1 => 'b', 2 => 'c']
```
Comment (drupol) [inline docs/pages/api.rst:171]:
```suggestion
            ->all(); // [0 => 'c', 1 => 'b', 2 => 'd']
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: docs/pages/api.rst (hunk 3 above)
Comment (AlexandruGG) [reply docs/pages/api.rst:455]: that makes sense, I added a note [here](https://github.com/loophp/collection/pull/84/commits/21bdbe8bf781d00c5078dcde232e1ae3c48b65e7#diff-89f6324da13155a697ae0b42136e03a40892dd4edc4257ea1e451f0ea2cc3f80R148-R150) to highlight this a bit better and added to the [contains example](https://github.com/loophp/collection/pull/84/commits/21bdbe8bf781d00c5078dcde232e1ae3c48b65e7#diff-89f6324da13155a697ae0b42136e03a40892dd4edc4257ea1e451f0ea2cc3f80R457-R459)

**Review by drupol — APPROVED** [review]
Thanks for taking the time to do this, I'm very grateful !

**Comment by AlexandruGG** [comment]
> Thanks!
> 
> I made a review. All those changes are great!
> 
> There's a small things to update, as soon as it's ok, I merge this.

Thank you, agreed the suggested change to that example makes it clearer 👍

**Comment by AlexandruGG** [comment]
> Thanks for taking the time to do this, I'm very grateful !

I appreciate all the work that you've already put in this project! 💯

**Comment by drupol** [comment]
Thanks again :)

**Pull request merged**